### DISCUSSION TITLE
Certain collection assertions seem to clear custom AssertionScope expectation (set with .WithExpectation)

### DISCUSSION THREAD

**Discussion by aracg** [fluentassertions/fluentassertions#2451] [Q&A] [answered]
I noticed that when I add a custom expectation to AssertionScope that contains certain collection assertions (like `NotBeEmpty()` or `HaveCountGreaterThan()`), the custom expectation is not present in failure messages. 

Is it a bug?

```cs
var inputSpecIds = new List<string>
{
    "09f0c79e-a206-4485-b316-04ea6603273b",
    "2e2522c6-f05e-4491-bb53-f5b7c78cdf96"
};

using (var testScope = new AssertionScope())
{
    testScope.WithExpectation("Some expectation details: ");
    inputSpecIds.Should().NotBeEmpty();
    inputSpecIds.Should().HaveCountGreaterThan(4);
    1.Should().Be(2);
}
```

```
Expected collection to contain more than 4 item(s), but found 2: {"09f0c79e-a206-4485-b316-04ea6603273b", "2e2522c6-f05e-4491-bb53-f5b7c78cdf96"}.
Expected value to be 2, but found 1.
```

**Comment by dennisdoomen** [comment]
That's because methods like `NotBeEmpty` set-up the expectation themselves and thereby overwrite whatever you passed. `WithExpectation` is supposed to be used in a single call to `Execute`.

**Reply by aracg** [reply]
Thanks for answering. That's what I guessed when I looked into the source, but wasn't sure it's intended.
I wasn't expecting it to work with some assertion methods, but not others - on the outside, there's no way to tell that they're different.

Maybe it would be helpful to mention this behavior in `WithExpectation()` documentation?

**Comment by dennisdoomen** [comment, marked as answer]
> Maybe it would be helpful to mention this behavior in WithExpectation() documentation?

It applies to the entire chain of `Execute`. I'm working on making this even more obvious. See also #2253
